### PR TITLE
Fix pipetrigger.close() to close the right file descriptors

### PR DESCRIPTION
**Pull request by mgedmin**
Here's what was happening:

- `pipetrigger.__init__()` creates two file descriptors by invoking os.pipe(), let's call them r1 and w1
- `asyncore.file_dispatcher.__init__()` wraps r1 in a `file_wrapper()` that does an `os.dup()`, creating r2
- `pipetrigger.__init__()` closes r1
- `pipetrigger.__init__()` keeps `self._fd = (r1, w1)` despite r1 being closed already!
- `pipetrigger._close()` closes all desciptors from `self._fd`, closing r1 again (causing an OSError for EBADF or closing a random real file the program still wants to use), but doesn't close self.socket, which is the file_wrapper instance holding r2 open.

I also had to fix one test that was calling trigger.close() and expecting trigger.recv() to keep working afterwards.

### CODE REVIEW

File: src/zope/server/trigger.py [hunk 1]
```diff
@@ -158,10 +158,7 @@ def _close(self):
                 self.socket.close()
                 self.socket = None
             if self.trigger is not None:
-                try:
-                    os.close(self.trigger)
-                except OSError:
-                    pass
+                os.close(self.trigger)
```
Comment (jamadden) [inline src/zope/server/trigger.py]: Nice!

Comment (mgedmin) [reply src/zope/server/trigger.py]: Thank you coverage.py and coveralls.io, or I wouldn't have dared ;)

**Comment by mgedmin** [comment]
I'll wait a bit to give the other invited reviewers a chance to chime in, if they want.

**Comment by icemac** [comment]
Seems reasonable but I do not feel competent enough for an actual review.

**Comment by mgedmin** [comment]
Thank you for reminding that this PR existed ;)

**Pull request merged**